terragraph_planner/optimization/topology_operations.py

```diff
@@ -504,125 +503,104 @@ def compute_capex(topology: Topology, params: OptimizerParams) -> Capex:
     )
 
 
-def _get_max_tx_power_of_all_links(
-    topology: Topology,
-    max_eirp_dbm: Optional[float],
-) -> Dict[str, float]:
-    """
-    Get max tx power for links in the topology.
-    """
-    max_tx_power = {}
-    for link_id, link in topology.links.items():
-        if (
-            link.link_type == LinkType.ETHERNET
-            or link.tx_sector is None
-            or link.rx_sector is None
-        ):
-            max_tx_power[link_id] = -math.inf
-            continue
-
-        tx_data = topology.sites[link.tx_site.site_id].device
-        max_tx_power[link_id] = get_max_tx_power(
-            tx_sector_params=tx_data.sector_params,
-            max_eirp_dbm=max_eirp_dbm,
-        )
-    return max_tx_power
-
-
 def update_link_caps_with_sinr(
-    topology: Topology, maximum_eirp: Optional[float]
+    topology: Topology,
+    params: OptimizerParams,
+    max_iterations: int = MAX_LINK_BUDGET_ITERATIONS,
```

Comment:
Put an argument here is easier for test, but if we don't expect users to use it directly, I'd prefer
1. remove this out from the arg list as before
2. or pass in the parameter in the call site (I believe there's only one caller)